src/coins.h

```diff
@@ -230,6 +230,31 @@ class CCoinsViewCursor
     uint256 hashBlock;
 };
 
+/**
+ * Cursor for iterating over the linked list of flagged entries in CCoinsViewCache.
+ */
+struct CoinsViewCacheCursor
+{
+    CoinsViewCacheCursor(size_t& usage LIFETIMEBOUND, CoinsCachePair& sentinel LIFETIMEBOUND, CCoinsMap& map LIFETIMEBOUND, bool will_erase) noexcept
+        : m_usage(usage), m_sentinel(sentinel), m_map(map), m_will_erase(will_erase) {}
+
+    inline CoinsCachePair* Begin() const noexcept { return m_sentinel.second.Next(); }
+    inline CoinsCachePair* End() const noexcept { return &m_sentinel; }
+
+    inline CoinsCachePair* Next(CoinsCachePair& current) noexcept
+    {
+        const auto next_entry{current.second.Next()};
+        return next_entry;
+    }
```

Comment:
Any reason not returning directly like in `Begin()`?
```suggestion
    inline CoinsCachePair* Next(CoinsCachePair& current) noexcept { return current.second.Next(); }
```

Reply:
It was just easier to split it up into 2 commits this way.

Reply:
Fixed.

Reply:
Ah, my mistake, this was an intermediary commit - fine either way